packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -134,21 +134,21 @@ export class SocketModeClient extends EventEmitter {
 
   /**
    * Start a Socket Mode session app.
-   * It may take a few milliseconds before being connected.
-   * This method must be called before any messages can be sent or received.
+   * This method must be called before any messages can be sent or received,
+   * or to disconnect the client via the `disconnect` method.
    */
   public start(): Promise<AppsConnectionsOpenResponse> {
     this.logger.debug('Starting a Socket Mode client ...');
     // Delegate behavior to state machine
     this.stateMachine.handle(Event.Start);
     // Return a promise that resolves with the connection information
     return new Promise((resolve, reject) => {
-      this.once(ConnectingState.Authenticated, (result) => {
+      this.once(State.Connected, (result) => {
```

Comment:
@seratch This is a change I just introduced that fixes the issue I mentioned. Now consumers of this package can run:

```
await client.start();
await client.disconnect();
```

.. without issue. Previously, since `start()` completed while the client was still connecting - that is, the websocket connection was not established yet - the above code would cause a "socket already closed" exception to be raised.

However, I understand this may be a significant behaviour change, though it feels like a positive one to me. What do you think?

Reply:
@filmaj This looks a nice improvement. Indeed, the behavior could be changed but it's favorable to almost 100% users. The only downside I can think of would be a slightly longer bootstrapping time (perhaps, 1-2 seconds?) and it may make deployment time longer accordingly. However, it's not 10+ seconds or minutes, so it should be totally fine for anyone. 

If you want to be as careful as possible about the behavior change, one possible solution would be to provide an opt-out option such as asyncStart: true (default value can be false, which means await for Connected state; true means waiting for Authenticated) for existing users. I don't think we should provide such, though.

Reply:
I think I will release as-is. People can always use the previous version if this behaviour is a show-stopper.